src/ApiService/ApiService/onefuzzlib/WebhookOperations.cs

```diff
@@ -1,125 +1,135 @@
+using System.Net;
+
+namespace Microsoft.OneFuzz.Service;
+
+public interface IWebhookOperations
+{
+    Async.Task SendEvent(EventMessage eventMessage);
+    Async.Task<Webhook?> GetByWebhookId(Guid webhookId);
+    Async.Task<bool> Send(WebhookMessageLog messageLog);
+}
+
+public class WebhookOperations : Orm<Webhook>, IWebhookOperations
+{
+
+    // Needs to eventually pull from global __version__
+    const string USER_AGENT = "onefuzz-webhook 0.0.0";
+    private readonly IWebhookMessageLogOperations _webhookMessageLogOperations;
+    private readonly ILogTracer _log;
+    private ICreds _creds;
+    private readonly IContainers _containers;
+
+    public WebhookOperations(ICreds creds, IStorage storage, IWebhookMessageLogOperations webhookMessageLogOperations, IContainers containers, ILogTracer log)
+        : base(storage, log)
+    {
+        _webhookMessageLogOperations = webhookMessageLogOperations;
+        _log = log;
+        _creds = creds;
+        _containers = containers;
+    }
+
+    async public Async.Task SendEvent(EventMessage eventMessage)
+    {
+        await foreach (var webhook in GetWebhooksCached())
+        {
+            if (!webhook.EventTypes.Contains(eventMessage.EventType))
+            {
+                continue;
+            }
+            await AddEvent(webhook, eventMessage);
+        }
+    }
+
+    async private Async.Task AddEvent(Webhook webhook, EventMessage eventMessage)
+    {
+        var message = new WebhookMessageLog(
+             EventId: eventMessage.EventId,
+             EventType: eventMessage.EventType,
+             Event: eventMessage.Event,
+             InstanceId: eventMessage.InstanceId,
+             InstanceName: eventMessage.InstanceName,
+             WebhookId: webhook.WebhookId
+            );
+
+        var r = await _webhookMessageLogOperations.Replace(message);
+        if (!r.IsOk)
+        {
+            var (status, reason) = r.ErrorV;
+            _log.Error($"Failed to replace webhook message log due to [{status}] {reason}");
+        }
+    }
+
+    public async Async.Task<bool> Send(WebhookMessageLog messageLog)
+    {
+        var webhook = await GetByWebhookId(messageLog.WebhookId);
+        if (webhook == null || webhook.Url == null)
+        {
+            throw new Exception($"Invalid Webhook. Webhook with WebhookId: {messageLog.WebhookId} Not Found");
+        }
+
+        var (data, digest) = await BuildMessage(webhookId: webhook.WebhookId, eventId: messageLog.EventId, eventType: messageLog.EventType, webhookEvent: messageLog.Event, secretToken: webhook.SecretToken, messageFormat: webhook.MessageFormat);
+
+        var headers = new Dictionary<string, string> { { "Content-type", "application/json" }, { "User-Agent", USER_AGENT } };
+
+        if (digest != null)
+        {
+            headers["X-Onefuzz-Digest"] = digest;
+        }
+
+        var client = new Request();
+        var response = client.Post(url: webhook.Url, json: data, headers: headers);

```

Comment:
Are you missing an `await` here?

```suggestion
        var response = await client.Post(url: webhook.Url, json: data, headers: headers);
```